File: skilloper_app/lib/constants/limits.dart
// Centralized limits for questionnaire validation
// Keep in sync with Go API limits in internal/models/limits.go

class QuizLimits {
  QuizLimits._(); // Prevent instantiation

  // Options per question
  static const int minOptions = 2;
  static const int maxOptions = 8;
  static const int defaultMaxOptions = 4;

  // Questions per questionnaire
  static const int maxQuestionsPerQuiz = 500;

  // Field lengths
  static const int maxTitleLength = 255;
  static const int maxDescriptionLength = 1000;

  // Alternative fields (for question variety)
  static const int maxAlternativeQuestions = 10;
  static const int maxAlternativeOptions = 20;
}

// Question type constants
// Keep in sync with Go API constants in internal/models/questionnaire.go
class QuestionTypes {
  QuestionTypes._(); // Prevent instantiation

  static const String singleChoice = 'single_choice';
  static const String multipleChoice = 'multiple_choice';
}

File: skilloper-api/internal/models/limits.go
package models

const (
	MinOptionsLimit   = 2
	MaxOptionsLimit   = 8
	DefaultMaxOptions = 4

	MaxQuestionsPerQuiz = 500

	MaxTitleLength       = 255
	MaxDescriptionLength = 1000

	MaxAlternativeQuestions = 10
	MaxExtraOptions         = 20

	StaleAttemptHours = 24

	ScorePercentage = 100

	MaxImportFileSize  = 10 * 1024 * 1024
	MaxRequestBodySize = 1 * 1024 * 1024
)

const (
	FormatJSON = "json"
)
